### PR TITLE
update

### PR DESCRIPTION
**Pull request by Piping**

### CODE REVIEW

**Comment by t-wissmann** [comment]
Hm? What is this PR about? It seems like a combination of multiple things and I don't see what the purpose of the changes are. Did you submit this PR by accident?

**Comment by Piping** [comment]
@t-wissmann oops. It is by accident. It was meant to be merged into my own fork right now.

**Comment by Piping** [comment]
@t-wissmann One question I have is how `vector<string> cmds` looks like ? I tried to use `mm->mouse_call_command` to invoke my command, but it does not work. For example, i tried `spwan xterm`, `xterm`, neither of them works.

File: src/xmainloop.cpp [hunk 1]
```diff
@@ -216,6 +216,10 @@ void XMainLoop::buttonpress(XButtonEvent* be) {
                     mm->mouse_initiate_move(client, {});
                 }
             }
+        } else {
+            // root windows handling
+            HSDebug("handle default RootWindow ButtonPress on win 0x%lx\n", be->window);
+            mm->mouse_call_command_root_window({"call", "spwan","xmenu.sh"});
```
Comment (t-wissmann) [inline src/xmainloop.cpp:222]: There's a typo `spawn` vs `spwan`. Also you're making a unencessary round-trip through the `mm` module. you can do the `callCommand` directly here:
```
std::ostringstream discardedOutput;
Input input("spawn", {"xmenu.sh"});
Command::call(input, discardedOutput);
```

**Comment by t-wissmann** [comment]
What does your xmenu.sh look like? Does it launch an Openbox/Fluxbox style menu on the root window?

**Comment by Piping** [comment]
image

Glad it works now on the build ( it might look ugly due to missing Xresource file in xephyr). Thanks for the correction. for the xmenu details, you can check out my script at 

`https://github.com/instantOS/instantUBUNTU/blob/dev/core/xmenu.sh`

**Pull request closed without merging**